app/assets/javascripts/components/features/compose/components/compose_form.jsx

```diff
@@ -174,7 +174,7 @@ class ComposeForm extends React.PureComponent {
 
           <div className='compose-form__publish'>
             <div className='character-counter__wrapper'><CharacterCounter max={500} text={text} /></div>
-            <div className='compose-form__publish-button-wrapper'><Button text={publishText} onClick={this.handleSubmit} disabled={disabled || text.replace(/[\uD800-\uDBFF][\uDC00-\uDFFF]/g, "_").length > 500} block /></div>
+            <div className='compose-form__publish-button-wrapper'><Button text={publishText} onClick={this.handleSubmit} disabled={disabled || text.replace(/[\uD800-\uDBFF][\uDC00-\uDFFF]/g, "_").length > 500 || text.trim().length === 0} block /></div>
```

Comment:
It makes sense.
However, I think it makes confusing that the button with text is enabled, but do not request.
How about `(text.length !==0 && text.trim().length === 0)` ?